core/src/main/java/com/linecorp/armeria/client/circuitbreaker/HttpCircuitBreakerClientBuilder.java

```diff
@@ -0,0 +1,110 @@
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.circuitbreaker;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Function;
+
+import com.google.common.base.MoreObjects;
+
+import com.linecorp.armeria.client.HttpClient;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.Response;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+/**
+ * A builder for a {@link HttpClient} decorator that handles failures of HTTP requests
+ * based on circuit breaker pattern.
+ */
+@UnstableApi
+public abstract class HttpCircuitBreakerClientBuilder<CB>
+        extends AbstractCircuitBreakerClientBuilder<CB, HttpResponse> {
+    static final int DEFAULT_MAX_CONTENT_LENGTH = Integer.MAX_VALUE;
+    private final boolean needsContentInRule;
+    private final int maxContentLength;
+
+    /**
+     * Creates a new builder with the specified {@link CircuitBreakerRule}.
+     */
+    protected HttpCircuitBreakerClientBuilder(ClientCircuitBreakerGenerator<CB> defaultMapping,
+                                              CircuitBreakerRule rule) {
+        super(defaultMapping, requireNonNull(rule, "rule"));
+        needsContentInRule = false;
+        maxContentLength = 0;
+    }
+
+    /**
+     * Creates a new builder with the specified {@link CircuitBreakerRuleWithContent} and
+     * the specified {@code maxContentLength}.
+     */
+    protected HttpCircuitBreakerClientBuilder(ClientCircuitBreakerGenerator<CB> defaultMapping,
+                                              CircuitBreakerRuleWithContent<HttpResponse> ruleWithContent,
+                                              int maxContentLength) {
+        super(defaultMapping, requireNonNull(ruleWithContent, "ruleWithContent"));
+        needsContentInRule = true;
+        this.maxContentLength = maxContentLength;
+    }
+
+    /**
+     * Returns the {@code maxContentLength} which is required to determine a {@link Response}
+     * as a success or failure. This value is valid only if {@link CircuitBreakerRuleWithContent} is used.
+     */
+    protected final int maxContentLength() {
+        return maxContentLength;
+    }
+
+    /**
+     * Returns a newly-created {@link CircuitBreakerClient} based on the properties of this builder.
+     */
+    protected CircuitBreakerClient build(HttpClient delegate,
+                                         CircuitBreakerHandlerFactory<CB, HttpRequest> factory) {
+        if (needsContentInRule) {
+            return new CircuitBreakerClient(delegate, mapping(), ruleWithContent(), maxContentLength, factory);
+        }
+        return new CircuitBreakerClient(delegate, mapping(), rule(), factory);
+    }
+
+    /**
+     * Returns a newly-created {@link CircuitBreakerClient} based on the properties of this builder.
+     */
+    public abstract CircuitBreakerClient build(HttpClient delegate);
```

Comment:
Question: Are you thinking of using `CircuitBreakerClient` also for the resilience4j circuit breaker? As you know we don't prefix `Http` to an HTTP client or decorator and only add `Rpc` to Thrift clients. If resilience4j circuit breaker uses `CircuitBreakerClient`, I slightly prefer consolidating this class with `CircuitBreakerClientBuilder`